src/kernel/crane_constructor/hook_chooser/hook.rs

```diff
@@ -0,0 +1,179 @@
+use crate::kernel::storage::storage::Storage;
+use crate::kernel::storage::storage::Value;
+use crate::Param_to_compare;
+use std::io;
+pub struct Hook{
```

Comment:
Don't forget about formating
Use cargo instruments to improve your code stile
- cargo [clippy](https://doc.rust-lang.org/clippy/usage.html)
- cargo [fmt](https://doc.rust-lang.org/book/appendix-04-useful-development-tools.html) --check
```rust
- pub struct Hook{
+ pub struct Hook {

```